### ISSUE TITLE
'support' release toolchain on all posixish platforms

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4449]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#4449|
|Type|enhancement|
|Created|2010-05-22 09:07:59Z|
|Branch|https://github.com/twisted/twisted/tree/non-linux-release-toolchain-4449|

Since I'm using MacOS a lot more these days, it would be nice if the release infrastructure tests ran there.  They seem to work fine, so I recommend we apply this patch:
```
#!diff
Index: twisted/python/test/test_release.py
===================================================================
--- twisted/python/test/test_release.py (revision 28979)
+++ twisted/python/test/test_release.py (working copy)
@@ -41,7 +41,7 @@
 from twisted.python._release import ChangeVersionsScript, BuildTarballsScript
 from twisted.python._release import NewsBuilder
 
-if sys.platform != 'linux2':
+if os.name != 'posix':
     skip = "Release toolchain only supported on Linux."
 else:
     skip = None
```

<details><summary>Searchable metadata</summary>

```
trac-id__4449 4449
type__enhancement enhancement
reporter__glyph glyph
priority__normal normal
milestone__ 
branch__branches_non_linux_release_toolchain_4449 branches/non-linux-release-toolchain-4449
branch_author__glyph__exarkun glyph, exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__tests tests
time__1274519279000000 1274519279000000
changetime__1274815016000000 1274815016000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="TimAllen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|TimAllen set owner to @glyph|
|-|-|

At the very least, the text of the skip message should be updated, too.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [29053]) Branching to 'non-linux-release-toolchain-4449'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [29054]) Only skip if os.name is not posix

refs [#4449](https://github.com/twisted/twisted/issues/4449)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun removed owner|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/non-linux-release-toolchain-4449)

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph set owner to @exarkun|
|-|-|

Thanks, much appreciated.  Looks good, land it.

**Comment by twisted-trac** [comment]
|<img alt="ivank's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|ivank commented|
|-|-|

Needs a NEWS file, no?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

> Needs a NEWS file, no?

Yep, thanks.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set status to `closed`|
|-|-|

(In [29056]) Merge non-linux-release-toolchain-4449

Author: glyph, exarkun
Reviewer: glyph, ivank
Fixes: [#4449](https://github.com/twisted/twisted/issues/4449)

Change the platform skip in the tests for the release tools so that only
POSIX is required, not Linux (various tests will still be skipped if other
requirements are not met, such as LaTeX not being installed).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
